Assets/VRTK/Scripts/Locomotion/VRTK_TouchpadMovement.cs

```diff
@@ -0,0 +1,480 @@
+        public bool flipBlink = true;
+
+        [Header("Horizontal Axis")]
+        [Tooltip("Selects the movement type to be performed when the horizontal axis changes occur.")]
+        public HorizontalAxisMovement horizontalAxisMovement = HorizontalAxisMovement.Slide;
+        [Tooltip("Dead zone for the horizontal axis. High value recommended for snap rotate and warp movement.")]
+        [Range(0, 1)]
+        public float horizontalDeadzone = 0.2f;
+        [Tooltip("The delay before the next snap rotation is allowed to happen.")]
+        public float snapRotateDelay = 0.5f;
+        [Tooltip("The number of degrees to instantly rotate in to the given direction.")]
+        public float snapRotateAngle = 30;
+        [Tooltip("The maximum speed the play area will be rotated when the touchpad is being touched at the extremes of the axis. If a lower part of the touchpad axis is touched (nearer the centre) then the rotation speed is slower.")]
+        public float rotateMaxSpeed = 3;
+
+        [Header("Shared Axis Settings")]
+        [Tooltip("Blink effect duration multiplier for the movement delay, ie. 1.0 means blink transition lasts until the delay has expired and 0.5 means the effect has completed when half of the delay time is done.")]
+        [Range(0.1f, 1)]
+        public float blinkDurationMultiplier = 0.7f;
+        [Tooltip("The maximum speed the play area will be moved by sliding when the touchpad is being touched at the extremes of the axis. If a lower part of the touchpad axis is touched (nearer the centre) then the speed is slower.")]
+        public float slideMaxSpeed = 3f;
+        [Tooltip("The speed in which the play area slows down to a complete stop when the user is no longer touching the touchpad. This deceleration effect can ease any motion sickness that may be suffered.")]
+        public float slideDeceleration = 0.1f;
+        [Tooltip("The delay before the next warp is allowed to happen.")]
+        public float warpDelay = 0.5f;
+        [Tooltip("The distance to warp in to the given direction.")]
+        public float warpRange = 1;
+        [Tooltip("The maximum altitude change allowed for a warp to happen.")]
+        public float warpMaxAltitudeChange = 1;
+
+        private GameObject controllerLeftHand;
+        private GameObject controllerRightHand;
+        private Transform playArea;
+        private Vector2 touchAxis;
+        private float movementSpeed = 0f;
+        private float strafeSpeed = 0f;
+        private float blinkFadeInTime = 0;
+        private float lastWarp = 0;
+        private float lastFlip = 0;
+        private float lastSnapRotate = 0;
+        private CapsuleCollider bodyCollider;
+        private Transform headset;
+        private bool leftSubscribed;
+        private bool rightSubscribed;
+        private ControllerInteractionEventHandler touchpadAxisChanged;
+        private ControllerInteractionEventHandler touchpadUntouched;
+
+        private void Awake()
```

Comment:
methods should follow their accessibility, so public methods first, then protected then private.

So these private methods need moving below the protected methods.